src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -415,6 +415,16 @@ private static boolean isSingleLineLambda(DetailAST lambda) {
         final DetailAST block = lambda.getLastChild();
         if (block.getType() != TokenTypes.SLIST) {
             result = lambda.getLineNo() == block.getLineNo();
+            final DetailAST elist = block.getFirstChild().findFirstToken(TokenTypes.ELIST);
+            if (elist != null) {
+                result = result && block.getLineNo() == elist.getLastChild().getLineNo();
+                final DetailAST expr = elist.findFirstToken(TokenTypes.EXPR);
+                if (expr != null && expr.getLastChild().getChildCount() > 1) {
+                    result = result
+                            && expr.getLastChild().getLastChild().getLineNo()
+                            == expr.getLastChild().getLineNo();
```

Comment:
"==" should be intended further to right, as it is not the same level of execution as "&&"